### ISSUE TITLE
Error occurred when executing LivePortraitLoadMediaPipeCropper

### ISSUE THREAD

**Issue by FugueSegue** [kijai/ComfyUI-LivePortraitKJ#92]
I'm experienced with ComfyUI but I have never used LivePortrait before. I've only worked with still images with Stable Diffusion and I've never worked with video before. I decided to try LivePortrait and it's been very frustrating. I've searched for and followed along with all of the tutorials I can find. I'm certain there is something important that I've missed or I don't understand. I've installed all the custom nodes without problem. I've installed all the models that I'm aware of and placed them in the proper directories. I think I installed InsightFace properly but I have no idea if I did it properly. I'm aware that InsightFace is now optional. I'm trying the example image workflow and I get this error message:

```
Error occurred when executing LivePortraitLoadMediaPipeCropper:

DLL load failed while importing onnxruntime_pybind11_state: A dynamic link library (DLL) initialization routine failed.

File "C:\ai\comfyui\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\nodes.py", line 536, in crop
self.cropper = CropperMediaPipe(**cropper_init_config)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\cropper.py", line 91, in __init__
self.landmark_runner = LandmarkRunner(
^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\landmark_runner.py", line 28, in __init__
import onnxruntime
File "C:\ai\comfyui\python_embeded\Lib\site-packages\onnxruntime\__init__.py", line 57, in
raise import_capi_exception
File "C:\ai\comfyui\python_embeded\Lib\site-packages\onnxruntime\__init__.py", line 23, in
from onnxruntime.capi._pybind_state import ExecutionMode # noqa: F401
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\python_embeded\Lib\site-packages\onnxruntime\capi\_pybind_state.py", line 32, in
from .onnxruntime_pybind11_state import * # noqa
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by kijai** [comment]
Issue is in the onnxruntime install, it can be annoying to get to work in some cases, I haven't ran into that specific issue myself so I don't really know how to solve it.

However I've added alternative to that too which you could try, while insightface does require onnxruntime, there's now also possibility to use either MediaPipe or face_alignment with torch model instead of onnx model by selecting "torch_gpu":

![image](https://github.com/user-attachments/assets/bc387f91-15d8-414a-a25a-13a1de4a673a)

**Comment by FugueSegue** [comment]
Thank you for your prompt response. But I am still getting an error.

Just to be clear, I am using the example workflow named liveportrait_image_example_01.json.

I changed the landmarkrunner_onnx_device to torch_gpu in the LivePortrait Load MediaPipeCropper.

I do not see a LivePortrait Load FaceAlignment Cropper in this workflow.

This is my error message:

```
Error occurred when executing LivePortraitLoadMediaPipeCropper:

No module named 'onnx2torch'

File "C:\ai\comfyui\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\nodes.py", line 536, in crop
self.cropper = CropperMediaPipe(**cropper_init_config)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\cropper.py", line 98, in __init__
self.landmark_runner = LandmarkRunnerTorch(
^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\landmark_runner.py", line 102, in __init__
self.model = torch.load(ckpt_path).to(self.device)
^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1025, in load
return _load(opened_zipfile,
^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1446, in _load
result = unpickler.load()
^^^^^^^^^^^^^^^^
File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1439, in find_class
return super().find_class(mod_name, name)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by FugueSegue** [comment]
I should add that I am using ComfyUI portable. I merely renamed the directory to "comfyui".

**Comment by kijai** [comment]
> Thank you for your prompt response. But I am still getting an error.
> 
> Just to be clear, I am using the example workflow named liveportrait_image_example_01.json.
> 
> I changed the landmarkrunner_onnx_device to torch_gpu in the LivePortrait Load MediaPipeCropper.
> 
> I do not see a LivePortrait Load FaceAlignment Cropper in this workflow.
> 
> This is my error message:
> 
> ```
> Error occurred when executing LivePortraitLoadMediaPipeCropper:
> 
> No module named 'onnx2torch'
> 
> File "C:\ai\comfyui\ComfyUI\execution.py", line 151, in recursive_execute
> output_data, output_ui = get_output_data(obj, input_data_all)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\ComfyUI\execution.py", line 81, in get_output_data
> return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\ComfyUI\execution.py", line 74, in map_node_over_list
> results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\nodes.py", line 536, in crop
> self.cropper = CropperMediaPipe(**cropper_init_config)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\cropper.py", line 98, in __init__
> self.landmark_runner = LandmarkRunnerTorch(
> ^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\ComfyUI\custom_nodes\ComfyUI-LivePortraitKJ\liveportrait\utils\landmark_runner.py", line 102, in __init__
> self.model = torch.load(ckpt_path).to(self.device)
> ^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1025, in load
> return _load(opened_zipfile,
> ^^^^^^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1446, in _load
> result = unpickler.load()
> ^^^^^^^^^^^^^^^^
> File "C:\ai\comfyui\python_embeded\Lib\site-packages\torch\serialization.py", line 1439, in find_class
> return super().find_class(mod_name, name)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

Oh my bad, I wasn't aware it still needs onnx2torch, it should be added to requirements I suppose. You would need to do `pip install onnx2torch` in your python env

**Comment by kijai** [comment]
> I should add that I am using ComfyUI portable. I merely renamed the directory to "comfyui".

Didn't notice this, to install stuff on portable, it would be running this in the portable root folder:

`python_embeded\python.exe -m pip install onnx2torch`

**Comment by FugueSegue** [comment]
Kijai, I managed to get it working. I started with a fresh install of ComfyUI and followed the directions carefully. Perhaps you updated something since yesterday and it was fixed. However, I think it was probably an error on my part. A good night's sleep and a fresh cup of coffee can make a difference!

I think a major error I made was attempting to install the incorrect WHL file. I thought the version installed with ComfyUI was 3.10 when it was actually 3.11. This may or may not be related to the problems I was having. But it was definitely an error I made the last time I attempted to install these custom nodes.

I think this issue (which was probably a non-issue) is resolved. Thank for your attention.
